### DISCUSSION TITLE
mysqldump: Got error: 2026: "TLS/SSL error: self-signed certificate"

### DISCUSSION THREAD

**Discussion by diadal** [spatie/laravel-backup#1862] [General]
All services that utilize DB are working fine except `laravel-backup`

Exception message: The dump process failed with a none successful exitcode. Exitcode ======== 2: Misuse of shell builtins Output ====== Error Output ============ mysqldump: Got error: 2026: "TLS/SSL error: self-signed certificate" when trying to connect

Exception trace: #0 /usr/local/var/www/app-home/vendor/spatie/db-dumper/src/DbDumper.php(194): Spatie\DbDumper\Exceptions\DumpFailed::processDidNotEndSuccessfully(Object(Symfony\Component\Process\Process)) #1 /usr/local/var/www/app-home/vendor/spatie/db-dumper/src/Databases/MySql.php(161): Spatie\DbDumper\DbDumper->checkIfDumpWasSuccessFul(Object(Symfony\Component\Process\Process), '/usr/local/var/...') #2 /usr/local/var/www/app-home/vendor/spatie/laravel-backup/src/Tasks/Backup/BackupJob.php(293): Spatie\DbDumper\Databases\MySql->dumpToFile('/usr/local/var/...') #3 [internal function]: Spatie\Backup\Tasks\Backup\BackupJob->Spatie\Backup\Tasks\Backup{closure}(Object(Spatie\DbDumper\Databases\MySql), 'backup') #4 /usr/local/var/www/app-home/vendor/laravel/framework/src/Illuminate/Collections/Arr.php(605): array_map(Object(Closure), Array, Array) #5 /usr/local/var/www/app-home/vendor/laravel/framework/src/Illuminate/Collections/Collection.php(785): Illuminate\Support\Arr::map(Array, Object(Closure)) #6 /usr/local/var/www/app-home/vendor/spatie/laravel-backup/src/Tasks/Backup/BackupJob.php(257): Illuminate\Support\Collection->map(Object(Closure)) #7 /usr/local/var/www/app-home/vendor/spatie/laravel-backup/src/Tasks/Backup/BackupJob.php(196): Spatie\Backup\Tasks\Backup\BackupJob->dumpDatabases() #8 /usr/local/var/www/app-home/vendor/spatie/laravel-backup/src/Tasks/Backup/BackupJob.php(170): Spatie\Backup\Tasks\Backup\BackupJob->createBackupManifest() #9 /usr/local/var/www/app-home/vendor/spatie/laravel-backup/src/Commands/BackupCommand.php(72): Spatie\Backup\Tasks\Backup\BackupJob->run() #10 /usr/local/var/www/app-home/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Spatie\Backup\Commands\BackupCommand->handle() #11 /usr/local/var/www/app-home/vendor/laravel/framework/src/Illuminate/Container/Util.php(43): Illuminate\Container\BoundMethod::Illuminate\Container{closure}() #12 /usr/local/var/www/app-home/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(95): Illuminate\Container\Util::unwrapIfClosure(Object(Closure)) #13 /usr/local/var/www/app-home/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure)) #14 /usr/local/var/www/app-home/vendor/laravel/framework/src/Illuminate/Container/Container.php(690): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL) #15 /usr/local/var/www/app-home/vendor/laravel/framework/src/Illuminate/Console/Command.php(213): Illuminate\Container\Container->call(Array) #16 /usr/local/var/www/app-home/vendor/symfony/console/Command/Command.php(279): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle)) #17 /usr/local/var/www/app-home/vendor/laravel/framework/src/Illuminate/Console/Command.php(182): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle)) #18 /usr/local/var/www/app-home/vendor/spatie/laravel-backup/src/Commands/BaseCommand.php(29): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #19 /usr/local/var/www/app-home/vendor/symfony/console/Application.php(1047): Spatie\Backup\Commands\BaseCommand->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #20 /usr/local/var/www/app-home/vendor/symfony/console/Application.php(316): Symfony\Component\Console\Application->doRunCommand(Object(Spatie\Backup\Commands\BackupCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #21 /usr/local/var/www/app-home/vendor/symfony/console/Application.php(167): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #22 /usr/local/var/www/app-home/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(197): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #23 /usr/local/var/www/app-home/artisan(35): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #24 {main}

**Comment by danharper83** [comment]
I have just randomly started getting this error, did you find out what the issue was?

**Reply by chinmaypurav** [reply]
Same here; was working fine few hours ago

**Reply by chinmaypurav** [reply]
Hey @danharper83 

I was using MariaDb as mysql client binary in my Dockerfile and I faced this issue.

On changing that to MySQL, it started working as expected.

```diff
RUN install-php-extensions intl gd xsl \
-    && apt update && apt install -y default-mysql-client \
-    && rm -rf /var/lib/apt/lists/*
+    && apt-get update && apt-get install -y wget gnupg lsb-release debconf-utils \
+    && wget https://dev.mysql.com/get/mysql-apt-config_0.8.34-1_all.deb \
+    && DEBIAN_FRONTEND=noninteractive dpkg -i mysql-apt-config_0.8.34-1_all.deb \
+    && echo "mysql-apt-config mysql-apt-config/select-server select mysql-8.4" | debconf-set-selections \
+    && apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y mysql-community-client \
+    && rm -rf /var/lib/apt/lists/* mysql-apt-config_0.8.34-1_all.deb
```

**Comment by saendee-r** [comment]
I got same error since last two weeks. I think it somehow related to the issue which mariadb-client have been set option to enable ssl by default
>> https://github.com/laravel/framework/issues/54269

So, I fixed it by adding 'add_extra_option' to mysqldump config on file config/database.php

'connections' => [
...
   'mysql' => [
   ...
    'dump' => [
                'add_extra_option' =>'--skip-ssl' 
      ],
   
   ]

]

then use php artisan config:clear to clear cache
and everything work fine.

**Reply by danharper83** [reply]
I also did this 👍

**Reply by chinmaypurav** [reply]
Due to compatibilty issue, I am now strictly using mysql client only over MariaDB client;

**Reply by rossjcooper** [reply]
Worked for me thanks
